services/cd-service/pkg/service/batch.go

```diff
@@ -92,12 +93,37 @@ func ValidateApplication(
 	return nil
 }
 
+func (d *BatchServer) checkUserPermissions(user *auth.User, act *api.CreateEnvironmentLockRequest, application, action string) error {
```

Comment:
Ideally, this function should be part of the `GrpcContextReader`.
But we can do that in another PR.

Reply:
we'll do this later